cpp/src/lance/format/schema.cc

```diff
@@ -361,7 +361,7 @@ std::shared_ptr<Field> Field::Copy(bool include_children) const {
 }
 
 std::shared_ptr<Field> Field::Project(const std::shared_ptr<::arrow::Field>& arrow_field) const {
-  assert(name_ == arrow_field->name());
+  //  assert(name_ == arrow_field->name());
```

Comment:
Why remove this?

Reply:
oh, in a debug change, lemme revert this.